lib/quoil/cli.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Quoil.CLI do
```

Comment:
the responsibilities for `Quoil.CLI` seem confused.
- it's the entry point to the app
- it handles console output for help, and general error handling of bad args
- it runs the main business logic of the app.

Focus on each unit of code having one responsibility.

It probably makes sense to turn this into a module whose responsibility is delegating to other parts of the app
- move help/error handling and early termination into `Quoil.ArgsProcessor`
- create a new `Ping` module that contains the main business logic, and move the body of `run_ping` there